gm4_mountaineering/data/gm4_mountaineering/functions/climbing_effects/add_climb_effects.mcfunction

```diff
@@ -1,14 +1,17 @@
+# summon shulker in front and below player, but 1000 blocks up, facing towards them
+execute if entity @s[y_rotation=135..180] positioned ~ ~-1.5 ~-1 align xyz unless entity @e[type=shulker, tag=gm4_mountaineering_climb_assist, dx=0, dz=0, dy=0] unless block ~0.5 ~ ~0.5 minecraft:air run summon minecraft:shulker ~0.5 ~1000 ~0.5 {Color:7,Peek:20,AttachFace:2,Invulnerable:1b,NoAI:1b,NoGravity:1b,PersistenceRequired:1b,Silent:1b,Tags:[gm4_mountaineering_climb_assist],DeathLootTable:"minecraft:empty",ActiveEffects:[{Id:14,Duration:100000,ShowParticles:0b}]}
+execute if entity @s[y_rotation=-180..-135] positioned ~ ~-1.5 ~-1 align xyz unless entity @e[type=shulker, tag=gm4_mountaineering_climb_assist, dx=0, dz=0, dy=0] unless block ~0.5 ~ ~0.5 minecraft:air run summon minecraft:shulker ~0.5 ~1000 ~0.5 {Color:7,Peek:20,AttachFace:2,Invulnerable:1b,NoAI:1b,NoGravity:1b,PersistenceRequired:1b,Silent:1b,Tags:[gm4_mountaineering_climb_assist],DeathLootTable:"minecraft:empty",ActiveEffects:[{Id:14,Duration:100000,ShowParticles:0b}]}
+execute if entity @s[y_rotation=-135..-45] positioned ~1 ~-1.5 ~ align xyz unless entity @e[type=shulker, tag=gm4_mountaineering_climb_assist, dx=0, dz=0, dy=0] unless block ~0.5 ~ ~0.5 minecraft:air run summon minecraft:shulker ~0.5 ~1000 ~0.5 {Color:7,Peek:20,AttachFace:5,Invulnerable:1b,NoAI:1b,NoGravity:1b,PersistenceRequired:1b,Silent:1b,Tags:[gm4_mountaineering_climb_assist],DeathLootTable:"minecraft:empty",ActiveEffects:[{Id:14,Duration:100000,ShowParticles:0b}]}
+execute if entity @s[y_rotation=-45..45] positioned ~ ~-1.5 ~1 align xyz unless entity @e[type=shulker, tag=gm4_mountaineering_climb_assist, dx=0, dz=0, dy=0] unless block ~0.5 ~ ~0.5 minecraft:air run summon minecraft:shulker ~0.5 ~1000 ~0.5 {Color:7,Peek:20,AttachFace:3,Invulnerable:1b,NoAI:1b,NoGravity:1b,PersistenceRequired:1b,Silent:1b,Tags:[gm4_mountaineering_climb_assist],DeathLootTable:"minecraft:empty",ActiveEffects:[{Id:14,Duration:100000,ShowParticles:0b}]}
+execute if entity @s[y_rotation=45..135] positioned ~-1 ~-1.5 ~ align xyz unless entity @e[type=shulker, tag=gm4_mountaineering_climb_assist, dx=0, dz=0, dy=0] unless block ~0.5 ~ ~0.5 minecraft:air run summon minecraft:shulker ~0.5 ~1000 ~0.5 {Color:7,Peek:20,AttachFace:4,Invulnerable:1b,NoAI:1b,NoGravity:1b,PersistenceRequired:1b,Silent:1b,Tags:[gm4_mountaineering_climb_assist],DeathLootTable:"minecraft:empty",ActiveEffects:[{Id:14,Duration:100000,ShowParticles:0b}]}
+
+# teleport shulker down 700 blocks to bring them into position
+execute as @e[type=shulker, tag=gm4_mountaineering_climb_assist, y=1000, dy=500] at @s run teleport @s ~ ~-1000 ~
```

Comment:
```suggestion
execute positioned ~ ~1000 ~ as @e[type=shulker, tag=gm4_mountaineering_climb_assist] at @s run teleport @s ~ ~-1000 ~
```

Reply:
This still leads to some kind of strange behaviour - this command alone, as far as I can tell, has the shulkers continuously teleporting down (so they go past the player again). If I put in some kind of check (eg predicate for only entities above y=500) then they teleport to the same place, but don't seem to open properly and running into them in any way causes the player to get stuck.